sync_diff_inspector/chunk/chunk.go

```diff
@@ -0,0 +1,460 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunk
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"go.uber.org/zap"
+)
+
+const (
+	lt  = "<"
+	lte = "<="
+	gt  = ">"
+)
+
+type ChunkType int
+
+const (
+	Bucket ChunkType = iota + 1
+	Random
+	Limit
+	Others
+	Empty
+)
+
+// Bound represents a bound for a column
+type Bound struct {
+	Column string `json:"column"`
+	Lower  string `json:"lower"`
+	Upper  string `json:"upper"`
+
+	HasLower bool `json:"has-lower"`
+	HasUpper bool `json:"has-upper"`
+}
+
+// ChunkID is to identify the sequence of chunks
+type ChunkID struct {
+	TableIndex int `json:"table-index"`
+	// we especially treat random split has only one bucket
+	// which is the whole table
+	// range is [left, right]
+	BucketIndexLeft  int `json:"bucket-index-left"`
+	BucketIndexRight int `json:"bucket-index-right"`
+	ChunkIndex       int `json:"chunk-index"`
+	//  `ChunkCnt` is the number of chunks in this bucket
+	//  We can compare `ChunkIndex` and `ChunkCnt` to know
+	// whether this chunk is the last one
+	ChunkCnt int `json:"chunk-count"`
+}
+
+func GetInitChunkID() *ChunkID {
+	return &ChunkID{
+		TableIndex:       -1,
+		BucketIndexLeft:  -1,
+		BucketIndexRight: -1,
+		ChunkIndex:       -1,
+		ChunkCnt:         0,
+	}
+}
+
+func (c *ChunkID) Compare(o *ChunkID) int {
+	if c.TableIndex < o.TableIndex {
```

Comment:
also check -1 and 1 will reduce indent level

Reply:
Done.